salt/states/file.py

```diff
@@ -244,6 +245,71 @@
 _ACCUMULATORS_DEPS = {}
 
 
+def _get_accumulator_dir():
+    '''
+    Return the directory that accumulator data is stored in, creating it if it
+    doesn't exist.
+    '''
+    fn_ = os.path.join(__opts__['cachedir'], 'accumulator')
+    if not os.path.isdir(fn_):
+        # accumulator_dir is not present, create it
+        os.makedirs(fn_)
+    return fn_
+
+
+def _get_accumulator_filepath():
+    '''
+    Return accumulator data path, uses PID as filename for identify.
+    #TODO it needs a better thing to use as unique entity that different
+    among salt run.
+    '''
+    return os.path.join(_get_accumulator_dir(), str(os.getpid()))
+
+
+def _cleanup_old_accummulator_data():
+    '''
+    Only keep current accumulator data.
+    '''
+    for fn in os.listdir(_get_accumulator_dir()):
+        path = os.path.join(_get_accumulator_dir(), fn)
+        if path != _get_accumulator_filepath():
```

Comment:
test also that pid does not exist anymore, if you call a salt-call and a salt-master or something else linked to the same config exists on the same box, you can have a race condition where you would delete the other process accumulated data.